### DISCUSSION TITLE
Visualize point PMTiles layer

### DISCUSSION THREAD

**Discussion by jinha** [opengeos/leafmap#605] [Q&A] [answered]
I have a point PMTiles layer uploaded on my server: https://lidar.digitalforestry.org/vt/in/2022/IN2022-Address_Points.pmtiles
When I tried to visualize the point layer using add_pmtiles, it does not show anything. I checked the tutorial and it only showed examples for polygons and lines only. Is there anyway to visualize a point PMTiles layer?

**Comment by giswqs** [comment]
It should work. Make sure you use the correct the layer name. 

https://protomaps.github.io/PMTiles/?url=https%3A%2F%2Flidar.digitalforestry.org%2Fvt%2Fin%2F2022%2FIN2022-Address_Points.pmtiles#map=7.46/39.614/-86.797
![image](https://github.com/opengeos/leafmap/assets/5016453/047317b5-87cf-4ba8-aefc-637fff866d01)

**Reply by jinha** [reply]
I also checked with the PMTiles viewer and confirmed that it is working okay. The problem is that when I tried to load it using leafmap, it is not working. Below is the code I used. I also went to "foliumap" github repository to see if they have any point layer example, but it did not have any point layer example neither. 

"""
import leafmap.foliumap as leafmap
ap_url = 'https://lidar.digitalforestry.org/vt/in/2022/IN2022-Address_Points.pmtiles'
ap_metadata = leafmap.pmtiles_metadata(ap_url)
print(f"layer names: {ap_metadata['layer_names']}")
print(f"bounds: {ap_metadata['bounds']}")

m = leafmap.Map(center=[37, -88], zoom=6, height='800px')
m.add_basemap('CartoDB.DarkMatter')
m.add_basemap('Esri.WorldImagery', show=False)

ap_style = {
    "version": 8,
    "sources": {
        "ap_source": {
            "type": "vector",
            "url": "pmtiles://" + ap_url,
            "attribution": 'PMTiles',
        }
    },
    "layers": [
        {
            "id": "address_point",
            "source": "ap_source",
            "source-layer": "in2022_address_points",
            "type": "point",
            "paint": {"fill-color": "#3388ff", "fill-opacity": 0.5},
        },
    ],
}

m.add_pmtiles(
    ap_url, style=ap_style, name='Address Points', overlay=True, show=True, zoom_to_layer=False
)

m
"""

**Comment by giswqs** [comment, marked as answer]
Try this one. The style can be a mapbox style. 

```python
import leafmap.foliumap as leafmap

m = leafmap.Map()

style = {
    "version": 8,
    "sources": {
        "example_source": {
            "type": "vector",
            "url": "pmtiles://" + url,
            "attribution": 'PMTiles',
        }
    },
    "layers": [
        {
            "id": "in2022_address_points",
            "source": "example_source",
            "source-layer": "in2022_address_points",
            "type": "circle",
            'paint': {
                'circle-radius': 3,
                'circle-color': '#0000ff'
            }
        },
    ],
}

m.add_pmtiles(
    url, name='PMTiles', style=style, overlay=True, show=True, zoom_to_layer=True
)
m
```
![image](https://github.com/opengeos/leafmap/assets/5016453/ab3bf927-f084-43a8-a713-21b2d858d0f4)

**Reply by giswqs** [reply]
More styles available here: https://docs.mapbox.com/style-spec/reference/layers/#circle

**Reply by jinha** [reply]
Thanks. It is working now. 😄
